enaml/core/parser.py

```diff
@@ -3445,6 +3445,56 @@ def p_arglist14(p):
     p[0] = Arguments(keywords=keywords, starargs=p[2], kwargs=p[8])
 
 
+def _validate_arglist_and_kwlist(p, items, keywords):
+    kwnames = set()
+    args = []
+    kws = []
+    _validate_arglist_list(items, p.lexer.lexer)
+    for arg in items:
+        if isinstance(arg, ast.keyword):
+            kws.append(arg)
+            kwnames.add(arg.arg)
+        else:
+            args.append(arg)
+    for kw in keywords:
+        if not isinstance(kw, ast.keyword):
+            msg = 'only named arguments may follow *expression'
+            tok = FakeToken(p.lexer.lexer, p.lineno(1))
```

Comment:
you might need to pass the `lineno` as an argument to this function. Only literal tokens like `STAR`, `COMMA`, etc. will correctly report line numbers. In all the rules where this is invoked, `p.lineno(1)` will  be an expression token, which will not report the correct line number. This is limitation in ply.